packages/node/src/login/oidc/refresh/TokenRefresher.spec.ts

```diff
@@ -24,44 +24,279 @@ import {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    const refreshedTokens = await refresher.refresh(
+      "mySession",
+      "some refresh token",
+      mockJwk()
+    );
+
+    expect(refreshedTokens.access_token).toEqual(mockDpopTokens().access_token);
+  });
 
-    const storageMock = mockStorageUtility(
-      {
-        [`solidClientAuthenticationUser:${key}`]: {
-          refreshToken: refreshTokenValue,
-        },
+  it("Refreshes a bearer token properly", async () => {
+    setupDefaultOidcClientMock();
+    const mockedStorage = mockStorageUtility({
+      "solidClientAuthenticationUser:mySession": {
+        issuer: "https://my.idp",
+        codeVerifier: "some code verifier",
+        redirectUri: "https://my.app/redirect",
+        dpop: "false",
       },
-      true
+    });
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    const refreshedTokens = await refresher.refresh(
+      "mySession",
+      "some refresh token"
     );
 
-    const tokenRefresher = getTokenRefresher({ storageUtility: storageMock });
-    await tokenRefresher.refresh(key);
-    expect(defaultMocks.tokenRequester.request).toHaveBeenCalledWith("global", {
-      /* eslint-disable camelcase */
-      grant_type: "refresh_token",
-      refresh_token: refreshTokenValue,
-      /* eslint-enable camelcase */
+    expect(refreshedTokens.access_token).toEqual(mockDpopTokens().access_token);
+  });
+
+  it("stores the refresh token if one is returned", async () => {
+    const mockedTokens = mockDpopTokens();
+    mockedTokens.refresh_token = "some new refresh token";
+    setupOidcClientMock(mockedTokens);
+
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
     });
+
+    const refreshedTokens = await refresher.refresh(
+      "mySession",
+      "some old refresh token",
+      mockJwk()
+    );
+
+    expect(refreshedTokens.refresh_token).toEqual("some new refresh token");
+
+    // Check that the session information is stored in the provided storage
+    await expect(
+      mockedStorage.getForUser("mySession", "refreshToken", { secure: true })
+    ).resolves.toEqual("some new refresh token");
+  });
+
+  it("throws if the IdP does not return an access token", async () => {
+    const mockedTokens = mockDpopTokens();
+    mockedTokens.access_token = undefined;
+    setupOidcClientMock(mockedTokens);
+
+    const mockedStorage = mockRefresherDefaultStorageUtility();
+
+    const refresher = getTokenRefresher({
+      storageUtility: mockedStorage,
+    });
+
+    await expect(
+      refresher.refresh("mySession", "some old refresh token", mockJwk())
+    ).rejects.toThrow("The IdP did not return the expected tokens.");
```

Comment:
Can we name the IdP (and are we consistently using "IdP" in our messages? I'd prefer "Identity Provider"), and can we list the 'expected tokens' that did not get returned?
```suggestion
    ).rejects.toThrow(`The Identity Provider [{$mockRefresherDefaultStorageUtility.getFromSession("mockRefresherDefaultStorageUtility").issuer}] did not return the expected tokens, expected [${expectedTokenList}].");
```

Reply:
I changed `IdP` to `Identity Provider` in our error messages, and specified the expected token list. on second thought, in the refresh flow, only an access token should be expected (since the ID token should not expire).

Reply:
Resolved in 8818797c031a7e51ec3c56e1bf2e195d54ff6dee